### ISSUE TITLE
[typescript-fetch] (4.0.X) Missing required parameter in <Model>FromJSON

### ISSUE THREAD

**Issue by someone1** [OpenAPITools/openapi-generator#1547]
##### Description
When there is a composition of multiple models, the generated typescript code does not add the required parameters of the extended type in the generated `FromJSON` function.

##### openapi-generator version
openapi-generator-cli-4.0.0-20181126.024631-44

##### OpenAPI declaration file content or url
```yaml
swagger: "2.0"
info:
  title: "Combined Test"

paths:
  "/get/Combined":
    get:
      summary: Retrieve List of Combined
      operationId: "get.Combined"
      responses:
        200:
          description: OK
          schema:
            $ref: "#/definitions/Combined"

definitions:         
  Simple:
    type: object
    required:
      - Name
    properties:
      Name:
        type: string
  Combined:
    allOf:
      - $ref: "#/definitions/Simple"
      - type: object
        properties:
          Occupation:
            type: string
```

Generated Code (excerpt):

```typescript
export interface Simple {
    /**
     * 
     * @type {string}
     * @memberof Simple
     */
    name: string;
}

export interface Combined extends Simple {
    /**
     * 
     * @type {string}
     * @memberof Combined
     */
    occupation?: string;
}

export function CombinedFromJSON(json: any): Combined {
    return {
        'occupation': !exists(json, 'Occupation') ? undefined : json['Occupation'],
    };
}
```

Compiler throws an error since its missing the required `name` parameter

##### Command line used for generation

openapi-generator generate -i openapi.yaml -l typescript-fetch -c es6.json -o ./typescript-api/

##### Steps to reproduce

1. Use the provided yaml excerpt above to generate a client with the provided command line
2. Buggy code generated

##### Related issues/PRs

PR #569

##### Suggest a fix/enhancement

Not sure - any suggestions on how to tackle this would be appreciated and I can add it to PR #1545 

Looking at the debugModels export for the provided yaml definition, it seems that `parentVars` is an empty list for model `Combined` which could have been used to add additional required params to the function. Alternatively, the `allVars` template variable could be used over the `vars` variable in this block as it contains all potential params?

**Comment by someone1** [comment]
I think this can also extend to the `ToJSON` generated function. All variables are not added to the returned JSON and no attention is given to `readOnly` variables. I think switching to the `allVars` template  variable makes sense for both functions and adding functionality for `readOnly` parameters can be added in the process.

**Comment by macjohnny** [comment]
cc @Place1

**Comment by Downchuck** [comment]
@someone1 Did you see this fixed? I'm seeing the same behavior in 4.0.0-beta2
